### PR TITLE
implement update program method.

### PR DESCRIPTION
**Pull request by Buwujiu**

### CODE REVIEW

File: src/main/java/org/icgc/argo/program_service/model/exceptions/NotFoundException.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package org.icgc.argo.program_service.model.exceptions;
```
Comment (d8660091) [inline src/main/java/org/icgc/argo/program_service/model/exceptions/NotFoundException.java]: We can use org.springframework.dao.EmptyResultDataAccessException

Comment (d8660091) [reply src/main/java/org/icgc/argo/program_service/model/exceptions/NotFoundException.java]: for more info https://github.com/icgc-argo/program-service/pull/72/files

File: src/main/java/org/icgc/argo/program_service/repositories/BaseRepository.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.repository.NoRepositoryBean;
+import org.springframework.data.repository.PagingAndSortingRepository;
+
+import java.util.Collection;
+import java.util.Set;
+
+@NoRepositoryBean
+public interface BaseRepository<T, ID> extends PagingAndSortingRepository<T, ID>, JpaSpecificationExecutor {
```
Comment (d8660091) [inline src/main/java/org/icgc/argo/program_service/repositories/BaseRepository.java]: PagingAndSortingRepository could be replaced with JpaRepository, it's more specific. And btw, this class looks stupid

File: src/test/java/org/icgc/argo/program_service/services/ProgramServiceTest.java [hunk 3]
```diff
@@ -78,9 +86,15 @@
...
   private MailService mailService;
 
+  @Mock
+  private PrimarySiteRepository primarySiteRepository;
+
+  private CommonConverter commonConverter;
+
   @BeforeEach
   void init() {
-    this.programService = new ProgramService(invitationRepository, programRepository, programConverter, egoService, mailService);
```
Comment (d8660091) [inline src/test/java/org/icgc/argo/program_service/services/ProgramServiceTest.java]: Could be replaced with @InjectMocks

File: src/main/java/org/icgc/argo/program_service/utils/EntityService.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    val requestedIds = ImmutableSet.copyOf(ids);
+    val existingIds = entities.stream().map(IdentifiableEntity::getId).collect(toImmutableSet());
+    val nonExistingIds = difference(requestedIds, existingIds);
+
+    checkNotFound(
+            nonExistingIds.isEmpty(),
+            "Entities of entityType '%s' were not found for the following ids: %s",
+            resolveEntityTypeName(entityType),
+            COMMA.join(nonExistingIds));
```
Comment (d8660091) [inline src/main/java/org/icgc/argo/program_service/utils/EntityService.java]: String.join(", ", nonExistingIds)

File: src/main/java/org/icgc/argo/program_service/repositories/BaseRepository.java (hunk 2 above)
Comment (Buwujiu) [reply src/main/java/org/icgc/argo/program_service/repositories/BaseRepository.java]: sounds good!

File: src/test/java/org/icgc/argo/program_service/services/ProgramServiceTest.java (hunk 3 above)
Comment (Buwujiu) [reply src/test/java/org/icgc/argo/program_service/services/ProgramServiceTest.java]: will do

**Pull request merged**